documentation/smg/docs_discussion.md

```diff
@@ -0,0 +1,203 @@
+* Encourage people to update as they code, rather than as an afterthought
+* Keeps docs always in-line with the main build commit
+
+##### Cons:
+* May confuse builds/deploys (Could be mitigated through docs labels)
+* Disallows combination of Herbert/Horace/Entire PACE docs into one place.
+
+#### Submodule
+Use git submodules to have the docs as a sub-repo of Horace main
+
+##### Pros:
+* Still have the separation of repos
+* Docs exist in Horace repo and can be updated/held there
+
+###### Cons:
+* Requires knowledge to manage submodules
+* May require changing build scripts
+* Means that the docs will be up-to-date with latest docs not current Horace commit
+
+## Repo Builds
+
+### Current state
+#### Manual trigger
+Currently a docs build/deploy is manually triggered from the Jenkins pipeline, which will build and (on build success) deploy to a github.io accessible place.
+
+##### Pros:
+* Control over when/where docs get built
+* Little risk of accidental destruction of docs site due to manual version specification
+
+##### Cons:
+* Requires manual effort to say when to deploy
+* Could lead to desynchronisation of repo and site if care not taken 
+
+### Alternatives
+#### Build/Deploy on updated main
+On every PR merge new build, replacing current docs
+
+##### Pros:
+* Keep docs up-to-date when they are updated
+* No effort to update docs
+
+##### Cons:
+* May lead to damage if docs not verified properly (mitigated by multiple docs versions)
+* Difficult to check how branches would appear
+* Requires manually updating version number in Jenkins
+
+#### Build/Deploy on Horace release
+Tie docs deploy into the existing deploy pipeline such that on each major Horace release docs are released to the right tag
+
+##### Pros:
+* Fully automated
+* Robust against accidental docs damage
+
+##### Cons:
+* Dev docs may grow out of sync in time between major releases
+* Dev docs may be difficult to track
+
+## Docs versioning
+### Current state
+#### Version folder and stable folder
+Currently we have a folder for each version and stable both updated/overwritten on a deploy
+
+##### Pros:
+* Easy to track
+* Pre-release can be set up as a prVersion and cleaned up on release
+
+##### Cons:
+* Manual cleanup of PR
+* Extra effort to make dev changes public
+
+### Alternatives
+#### Unstable, stable, version
+Have an unstable (dev) folder which is updated every push, stable updated on release and version updated every push
+
+##### Pros:
+* Dev changes immediately update through unstable version
+* Stable version for users directs to most recent
+
+##### Cons:
+* More Jenkins logic required
+* Extra folders
+
+#### Dev, Version
+Only have version number and dev version users look up their version
+
+##### Pros:
+* Users of old versions have no confusion on release update
+* Dev changes visible
+
+##### Cons:
+* No short cut to latest version (need to know your version)
+* Discourages users staying up to date?
+
+#### Per-branch docs
```

Comment:
I feel per-branch docs are probably overkill, which I see you've noted below. My feeling is a set of docs for each release and a set of docs for master would be sufficient.